src/main/java/org/broadinstitute/consent/http/authentication/OAuthAuthenticator.java

```diff
@@ -67,42 +71,47 @@ private AuthUser buildAuthUserFromHeaders(Map<String, String> headers) {
    * Attempt to get the registration status of the current user and set the value on AuthUser
    *
    * @param authUser The AuthUser
-   * @return A cloned AuthUser with Sam registration status
+   * @return A cloned AuthUser with Sam registration status and a Consent User
    */
   private AuthUser getUserWithStatusInfo(AuthUser authUser) {
     if (authUser == null || authUser.getEmail() == null) {
       logWarn("AuthUser/email is null, cannot get user status info");
       return null;
     }
+    try {
+      authUser.setUser(userService.findUserByEmail(authUser.getEmail()));
+    } catch (Exception e) {
+      logException("Error finding Consent user: " + authUser.getEmail(), e);
+    }
     try {
       UserStatusInfo userStatusInfo = samService.getRegistrationInfo(authUser);
       if (Objects.nonNull(userStatusInfo)) {
         // safety check in case the call to generic user (i.e. Google) failed.
-        if (Objects.isNull(authUser.getEmail())) {
+        if (authUser.getEmail() == null) {
```

Comment:
If it's possible that the user email is `null` here, wouldn't we want to call `findUserByEmail()` after the Sam lookup?

Reply:
Yes, good point. I see that the comment here is pretty outdated as well - at one point in time, we queried Google for the user's information, but that isn't the case now.